articles/quickstart/spa/auth0-react/01.md

```diff
@@ -0,0 +1,297 @@
+
+With React Router, your goal is to use the `history` object within the `onRedirectCallback` method to navigate to that private route programmatically. React Router exposes the `useHistory()` custom React Hook to make it easy to access the `history` object in any component that is a descendant of the `BrowserRouter` component.
+
+As such, you can create an `Auth0ProviderWithHistory` function component that renders `Auth0Provider`. This wrapper component then calls `useHistory()` within its body to make the `history` object available within the scope of `Auth0Provider` and its  `onRedirectCallback` event handler.
+
+**Root component:**
+
+```javascript
+ReactDOM.render(
+  <BrowserRouter>
+    <Auth0ProviderWithHistory>
+      <App />
+    </Auth0ProviderWithHistory>
+  </BrowserRouter>,
+  document.getElementById("root")
+);
+```
+
+**`Auth0ProviderWithHistory` component:**
+
+```javascript
+const Auth0ProviderWithHistory = ({ children }) => {
+  const history = useHistory();
+
+  const onRedirectCallback = (appState) => {
+    history.push(appState?.returnTo || window.location.pathname);
+  };
+
+  return (
+    <Auth0Provider
+      domain="${account.namespace}"
+      client_id="${account.clientId}"
+      redirect_uri={window.location.origin}
+      onRedirectCallback={onRedirectCallback}
+    >
+      {children}
+    </Auth0Provider>
+  );
+};
+```
+
+The implementation of the authentication route guard may look like this `PrivateRoute` component:
+
+```javascript
+const PrivateRoute = ({ component, ...args }) => (
+  <Route component={withLoginRequired(component)} {...args} />
+);
+```
+
+You can use the `PrivateRoute` component in the same way that you use the `Route` component from React Router:
+
+```javascript
+function App() {
+  const { isLoading } = useAuth0();
+
+  return (
+    <div className="App">
+      {isLoading ? (
+        <h2>Loading...</h2>
+      ) : (
+        <>
+          <NavBar />
+          <Switch>
+            <Route path="/" exact component={Home} />
+            <PrivateRoute path="/profile" component={Profile} />
+          </Switch>
+        </>
+      )}
+    </div>
+  );
+}
+```
+
+⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️
+
+**TODO: Final URL to sample.**
+ 
+⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️
+
+Take a look at this sample that puts this technique together while also implementing a navigation bar: "Using React Router with the Auth0 React SDK" [sample required]
+
+## Call an API
+
+Your React application may need to access protected data from an API that you are also [securing with Auth0](https://auth0.com/docs/microsites/protect-api/protect-api). For that use case, you'll need to retrieve an access token in the format of a [JSON Web Token (JWT)](https://jwt.io/) to authorize any calls you make to that secured API. 
+
+For the Auth0 React SDK to configure the access token correctly, you need to pass an `audience` property to the `Auth0Provider` as follows:
```

Comment:
This page mentions nothing about the localhost issue & consent which you can bump into when using Auth0 from localhost. There should at least be some mention about this.

Reply:
@sandrinodimattia @stevehobbsdev I remember Andreas mentioning that the consent screen only comes up during localhost development. Is this considered a bug? Is this documented anywhere in the docs?

Reply:
It's not a bug, it's by design. However, we need to provide customers with a path to solve it if they bump into the issue. A link to some page will help here. We want to avoid content like this showing up because we're not documenting this good enough: https://dev.to/kleeut/solving-consentrequired-auth0-spa-sdk-1le2

Reply:
> https://dev.to/kleeut/solving-consentrequired-auth0-spa-sdk-1le2
This issue will be resolved by https://github.com/auth0/auth0-spa-js/pull/482 (the react sdk will use `checkSession` when it's released and this will allow users to handle recoverable errors like `consent_required` by having the user login again)

Reply:
Moved to CF.